### PR TITLE
Add plugin for LeakCanary 2

### PR DESCRIPTION
**Pull request by hbmartin**
## Summary

Add plugin for LeakCanary 2 as requested various times: https://github.com/facebook/flipper/issues/1379 https://github.com/facebook/flipper/issues/832  https://github.com/square/leakcanary/issues/1777

## Changelog

* Adds a leakcanary2 plugin for Android
* Adds support for leakcanary2 to existing desktop plugin

## Test Plan

* Docs updated to show new implementation
* Should old leakcanary plugin in sample be replaced?

### CODE REVIEW

**Comment by hbmartin** [comment]
@passy Look forward to your feedback! I'm signing the CLA now.

**Comment by passy** [comment]
Wow! Amazing! 🚀

I'm not sure I'll have time to fully review this today, but I'm super excited about this. Thank you!

**Comment by bizzguy** [comment]
This is awesome.  Thanks!!

File: docs/setup/leak-canary-2-plugin.mdx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
```
Comment (passy) [inline docs/setup/leak-canary-2-plugin.mdx:1]: This is awesome! Thanks for adding docs with this! :)

File: docs/setup/leak-canary-2-plugin.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      /*
+      add leak canary plugin to flipper
+      */
+      client.addPlugin(LeakCanary2FlipperPlugin())
+      client.start()
+    }
+  }
+```
+
+That's it!
```
Comment (passy) [inline docs/setup/leak-canary-2-plugin.mdx:49]: 🥳

**Review by passy — APPROVED** [review]
Looks great! I'm not sure if there are any style guides we enforce internally but if there's anything, I'll update it accordingly.

**Pull request closed without merging**